File: frontend/ANGULAR/src/app/services/models/product-response.ts
/* tslint:disable */
/* eslint-disable */
export interface ProductResponse {
  category?: 'ELETTRONICA' | 'FINANZE' | 'INVESTIMENTI' | 'ASSICURAZIONI' | 'IMMOBILIARE' | 'LAVORO' | 'ECOMMERCE' | 'SERVIZI' | 'STARTUP' | 'CONTABILITA' | 'VIAGGI' | 'SPORT' | 'FITNESS' | 'SALUTE' | 'BELLEZZA' | 'MODA' | 'ALIMENTAZIONE' | 'AUTOMOTIVE' | 'CASA' | 'GIOIELLI' | 'LIBRI' | 'MUSICA' | 'FILM' | 'ARTE' | 'MEDITAZIONE' | 'TERAPIA' | 'MOTIVAZIONE' | 'SVILUPPO_PERSONALE' | 'BENESSERE_MENTALE' | 'SCRITTURA' | 'APP' | 'GIOCHI_BAMBINI' | 'EDUCAZIONE' | 'FAMIGLIA' | 'VACANZE' | 'CASA_FAMIGLIA' | 'SALUTE_FAMIGLIA' | 'TEMPO_LIBERO' | 'CUCINA' | 'FESTE';
  description?: string;
  discount?: number;
  finalPrice?: number;
  id?: number;
  image_url?: string;
  link?: string;
  name?: string;
  price?: number;
}

File: backend/src/main/java/com/leonardo/hug/product/ProductResponse.java
package com.leonardo.hug.product;

import com.leonardo.hug.category.Category;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ProductResponse {

    private Long     id;
    private String   name;
    private String   description;
    private Double   price;
    private Category category;
    private Double   discount;
    private String   image_url;
    private String   link;
    private Double   finalPrice;
}
